sisl/viz/plotly/input_fields/array.py

```diff
@@ -0,0 +1,42 @@
+import numpy as np
+
+from .._input_field import InputField
+
+
+class ArrayInput(InputField):
+
+    dtype = "array-like"
+
+    _type = 'array'
+
+    _default = {
+        "width": "s100% l50%"
+    }
+
+    def __init__(self, *args, **kwargs):
+
+        # If the shape of the array was not provided, register it
+        # This is important because otherwise when the setting is set to None
+        # there is no way of knowing how to display the input field.
+        # For variable shapes, a different input (ListInput) should be used
+        try:
+            kwargs["params"]["shape"]
+        except:
+            shape = np.array(kwargs["default"]).shape
+
+            if kwargs.get("params", False):
+                kwargs["params"]["shape"] = shape
+            else:
+                kwargs["params"] = {"shape": shape}
+
+        super().__init__(*args, **kwargs)
+
+
+class Array1dInput(ArrayInput):
+
+    _type = 'vector'
```

Comment:
are `_type` needed? Wouldn't `isinstance(obj, Array1dInput)` suffice?

Also, I would like to rename the classes to:
```python
ArrayNDInput
Array1DInput(ArrayNDInput)
Array2DInput(ArrayNDInput)
```
for clarity; this also clarifies that an array is really n-dimensional and can have any shape?

What do you think? I see that all other input fields also have the `_type` so perhaps it is ok. I just wonder the difference between `_type` and `isinstance`, is there a reason for the former?

Reply:
`_type` is a class variable that is used by the javascript side (the GUI) to understand how the input should be rendered. It's never used in python, where indeed you would use `isinstance` to check if an input field is of some kind. Maybe I could send the full `mro` of the input field and the GUI could decide how to render it, but I'm afraid this would significally slow down the rendering if it needs to be checked for each input.

Another approach could be to determine only once on launch how each input field should be rendered and somehow store it. But I don't know if this would have some drawbacks.

I will change `ArrayInput` to `ArrayNDInput`.

In general, the input field objects were just introduced to let the GUI know how to render input fields. Afterwards, I've been introducing features that I thought could be useful in python. But some features may not make sense if you think python-wise only :)

Reply:
Ok, then keep it!! :) I thought it was internal use only.